### ISSUE TITLE
Memory leak in spinner component

### ISSUE THREAD

**Issue by MsKrypt1** [Napster2210/ngx-spinner#111]
It looks like spinner produces memory leaks because of incorrect unsubscription in the component. I'm referring to this piece:
https://github.com/Napster2210/ngx-spinner/blob/978cf4aac7a17bdd88f9dd0a9b3e09c7eb31804c/projects/ngx-spinner/src/lib/ngx-spinner.component.ts#L203-L205

Unsubscribe doesn't stop the spinner service subscription. `takeWhile` operator waits until the provided subject emits a value and then unsubscribes. So it should be like this:
```js
  this.ngUnsubscribe.next();
  this.ngUnsubscribe.complete();
```

[A comparison in StackBlitz](https://stackblitz.com/edit/rxjs-unsubscription). Check the console messages, especially when component is destroyed.

Also just wanted to say, that awesome design! The spinners really do look great.

**Comment by rspl-yuvraj** [comment]
@MsKrypt1 Thanx a lot for pointing out the issue, I'll resolve this in the next release.

**Comment by rspl-yuvraj** [comment]
@MsKrypt1 This has been resolved and will be available in the next version.

**Comment by Napster2210** [comment]
This is resolved in version `v8.1.0`
